picoquic/frames.c

```diff
@@ -703,19 +703,45 @@ uint8_t* picoquic_decode_stream_frame(picoquic_cnx_t* cnx, uint8_t* bytes, const
     return bytes;
 }
 
+int stream_reset_sent(picoquic_stream_head* stream) {
+    return (stream->stream_flags & picoquic_stream_flag_reset_sent) != 0;
+}
+
+int stream_reset_requested(picoquic_stream_head* stream) {
+    return (stream->stream_flags & picoquic_stream_flag_reset_requested) != 0;
+}
+
+int stream_stop_sending_requested(picoquic_stream_head* stream) {
+    return (stream->stream_flags & picoquic_stream_flag_stop_sending_requested) != 0;
+}
+
+int stream_stop_sending_sent(picoquic_stream_head* stream) {
+    return (stream->stream_flags & picoquic_stream_flag_stop_sending_sent) != 0;
+}
+
+int stream_stop_sending_received(picoquic_stream_head* stream) {
+    return (stream->stream_flags & picoquic_stream_flag_stop_sending_received) != 0;
+}
+
+int stream_fin_notified(picoquic_stream_head* stream) {
+    return (stream->stream_flags & picoquic_stream_flag_fin_notified) != 0;
+}
+
+int stream_fin_sent(picoquic_stream_head* stream) {
+    return (stream->stream_flags & picoquic_stream_flag_fin_sent) != 0;
+}
+
```

Comment:
This is very similar to the work that @igorlord did to "Simplify stream id tests" in PR #249. The difference is that Igor used macros instead of C functions. We can argue that compilers can inline the C functions just like they can with macros, and the there are no performance differences. But there is an impact on name spaces. The globally defined functions become visible in the picoquic lib, and their names can conflict with similar names used for example by the application that handles the streams. In contrast, the macros are defined in the "internal" header, are not visible by the application, and thus will not create name space collisions.

I think there are three options: use macros, use a picoquic_ prefix in the name of functions, or define the functions as "static". What do you prefer?